web3-wrapper/src/web3_wrapper.ts

```diff
@@ -715,19 +714,23 @@ export class Web3Wrapper {
      * @return The contents nested under the result key of the response body
      */
     public async sendRawPayloadAsync<A>(payload: Partial<JSONRPCRequestPayload>): Promise<A> {
-        const sendAsync = this._provider.sendAsync.bind(this._provider);
+        if (!payload.method) {
+            throw new Error(`Must supply method in JSONRPCRequestPayload, tried: [${payload}]`);
+        }
+        // tslint:disable:no-object-literal-type-assertion
         const payloadWithDefaults = {
             id: this._jsonRpcRequestId++,
             params: [],
             jsonrpc: '2.0',
             ...payload,
-        };
-        const response = await promisify<JSONRPCResponsePayload>(sendAsync)(payloadWithDefaults);
-        if (response.error) {
-            throw new Error(response.error.message);
+        } as JSONRPCRequestPayload;
+        // tslint:enable:no-object-literal-type-assertion
+        const sendAsync = promisify(this._provider.sendAsync.bind(this._provider));
+        const response = await sendAsync(payloadWithDefaults); // will throw if it fails
+        if (!response || !response.result) {
+            throw new Error(`No response`);
```

Comment:
I think we really want to throw the `response.error.message` if it's available to us.